src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector256_1.cs

```diff
@@ -173,17 +185,46 @@ internal static void ThrowIfUnsupportedType()
         /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
         public bool Equals(Vector256<T> other)
         {
-            ThrowIfUnsupportedType();
-
-            for (int i = 0; i < Count; i++)
+            if (Avx.IsSupported)
             {
-                if (!((IEquatable<T>)(GetElement(i))).Equals(other.GetElement(i)))
+                if (typeof(T) == typeof(float))
+                {
+                    Vector256<float> result = Avx.Compare(AsSingle(), other.AsSingle(), FloatComparisonMode.EqualOrderedNonSignaling);
+                    return Avx.MoveMask(result) == 0b1111_1111;
+                }
+
+                if (typeof(T) == typeof(double))
                 {
-                    return false;
+                    Vector256<double> result = Avx.Compare(AsDouble(), other.AsDouble(), FloatComparisonMode.EqualOrderedNonSignaling);
+                    return Avx.MoveMask(result) == 0b1111;
                 }
             }
 
-            return true;
+            if (Avx2.IsSupported)
+            {
+                // Unlike float/double, there are no special values to consider
+                // for integral types and we can just do a comparison that all
+                // bytes are exactly the same.
+
+                Debug.Assert((typeof(T) != typeof(float)) && (typeof(T) != typeof(double)));
+                Vector256<byte> result = Avx2.CompareEqual(AsByte(), other.AsByte());
+                return Avx2.MoveMask(result) == unchecked((int)(0b1111_1111_1111_1111_1111_1111_1111_1111));
```

Comment:
Maybe?
```suggestion
                return Avx2.MoveMask(result) == unchecked((int)(0xFFFF_FFFF));
```

Then I would do the same on other places, as the constants aren't so long and easier to read (but I think it's more a personal preference?).

Reply:
I had done it this way because we have 1 bit per element, and I thought it better to explicitly call out the bits here.

Reply:
Added a comment clarifying.

Reply:
Perfect 👍 Thx!